lib/src/fix.rs

```diff
@@ -58,6 +58,29 @@ pub struct FileToFix {
     pub repo_path: RepoPathBuf,
 }
 
+/// The results of applying the [FileFixer] to fix a given file.
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct FixResult {
+    /// The FileId of the file after applying the fixer. This will be a new
+    /// FileId if the contents changed, or the same as before if nothing
+    /// changed.
+    pub file_id: FileId,
+    /// Could be empty, or one value for accumulated stderr output,
+    /// or multiple values from a more structured source.
+    pub messages: Vec<FixerMessage>,
+}
```

Comment:
Nit: I liked the previous variant a bit more, since each state was represented separately. I think it will be worthwhile to move the inner state of the result into a enum in the future. 

For now, I think it'd be best if the `file_id` field was a `Option<FiledId>` so it models the unchanged part correctly and gives the caller a heads up that it did nothing.